### DISCUSSION TITLE
are both of this considered parameterized? when using queryRaw and queryRawUnsafe

### DISCUSSION THREAD

**Discussion by Cariaga** [prisma/prisma#18773] [Q&A] [answered]
```
    
const query_results = await prisma.$queryRaw(Prisma.sql
 `SELECT * from my_function(${mindate}::timestamp without time zone, ${maxdate}::timestamp without time zone,${adminlogsSearch.search}::text, ${ adminlogsSearch.size}::integer)`);
    
```

```
 const query_results2 = await prisma.$queryRawUnsafe(
 `SELECT * FROM my_function($1::timestamp without time zone, $2::timestamp without time zone, $3::text, $4::integer)`,
mindate, maxdate, search, size);
```

this just idea/suggestion  but i think this should be a thing if possible?
```
const query_results3 = await prisma.$queryRaw(Prisma.sql
      `SELECT * from my_function($1::timestamp without time zone, $2::timestamp without time zone,$3::text, $4::integer)`
      ,  mindate, maxdate, search, size);        
```

**Comment by ludralph** [comment, marked as answer]
Hi @Cariaga 👋 

The third approach that you suggested is the correct way to use parameterized queries with Prisma. This approach uses the $1, $2, $3, and $4 placeholders to indicate where the values of mindate, maxdate, search, and size should be substituted in the SQL query. The placeholders are then passed as arguments to the prisma.$queryRaw() function, after the SQL query string.

**Reply by Cariaga** [reply]
thank you ludralph,

the 3rd one is actually receiving this error when I use it 

```
Invalid `prisma.$queryRaw()` invocation:
Your raw query had an incorrect number of parameters. Expected: `4`, actual: `0`.
Error:
Invalid `prisma.$queryRaw()` invocation:
Your raw query had an incorrect number of parameters. Expected: `4`, actual: `0`.
```

the first two worked fine 

my package.json is
```
   "@prisma/client": "^4.12.0",
   "prisma": "^4.12.0",
```

**Reply by ludralph** [reply]
@Cariaga can you share your schema so I can try out the third one?

**Reply by Cariaga** [reply]
it may have been a previous version i had but this works now await prisma.$queryRaw(Prisma.sql``);

**Reply by ludralph** [reply]
Thanks for confirming 🙌 .
